examples/resume/README.md

```diff
@@ -0,0 +1,3 @@
+New state CANCELLING_BUT_RESUMING is introduced to support resume after cancellation
+
+![image](https://github.com/mvrsss/rmqcpp/assets/60746841/0ccd36db-8a7b-4f15-94fd-764d1559a73b)
```

Comment:
Let's put this image in this repo under the `docs` folder, and add a brief markdown file in there which references it. This README describes internals to the library but so far the examples folder is used for end-users to quickly get started. I think we should keep those purposes separate